DigitalLearningSolutions.Data.Tests/Services/UserServiceTests.cs

```diff
@@ -1,66 +1,106 @@
+    using DigitalLearningSolutions.Data.Models.User;
     using DigitalLearningSolutions.Data.Services;
     using DigitalLearningSolutions.Data.Tests.Helpers;
+    using FakeItEasy;
     using FluentAssertions;
     using NUnit.Framework;
 
     public class UserServiceTests
     {
         private IUserService userService;
+        private IUserDataService userDataService;
 
         [SetUp]
         public void Setup()
         {
-            var connection = ServiceTestHelper.GetDatabaseConnection();
-            userService = new UserService(connection);
+            userDataService = A.Fake<IUserDataService>();
+            userService = new UserService(userDataService);
         }
 
         [Test]
-        public void GetAdminUserByUsername_Returns_admin_user()
+        public void GetUsersByUsername_Returns_admin_user_and_delegate_users()
         {
             // Given
             var expectedAdminUser = UserTestHelper.GetDefaultAdminUser();
+            var expectedDelegateUsers = UserTestHelper.GetDefaultDelegateUser();
+            A.CallTo(() => userDataService.GetAdminUserByUsername(A<string>._)).Returns(UserTestHelper.GetDefaultAdminUser());
+            A.CallTo(() => userDataService.GetDelegateUsersByUsername(A<string>._)).Returns(new List<DelegateUser> { UserTestHelper.GetDefaultDelegateUser() });
 
             //When
-            var returnedAdminUser = userService.GetAdminUserByUsername("Username");
+            var (returnedAdminUser, _) = userService.GetUsersByUsername("Username");
+            var (_, returnedDelegateUsers) = userService.GetUsersByUsername("SV1234");
             returnedAdminUser.Id = expectedAdminUser.Id;
+            returnedDelegateUsers.First().Id = expectedDelegateUsers.Id;
 
             // Then
             returnedAdminUser.Should().BeEquivalentTo(expectedAdminUser);
+            returnedDelegateUsers.FirstOrDefault().Should().BeEquivalentTo(expectedDelegateUsers);
         }
 
         [Test]
-        public void GetDelegateUsersByUsername_Returns_delegate_users()
+        public void GetUsersById_Returns_admin_user_and_delegate_user()
         {
             // Given
-            var expectedDelegateUsers = UserTestHelper.GetDefaultDelegateUser();
+            var expectedAdminUser = UserTestHelper.GetDefaultAdminUser();
+            var expectedDelegateUser = UserTestHelper.GetDefaultDelegateUser();
+            A.CallTo(() => userDataService.GetAdminUserById(A<int>._)).Returns(UserTestHelper.GetDefaultAdminUser());
+            A.CallTo(() => userDataService.GetDelegateUserById(A<int>._)).Returns(UserTestHelper.GetDefaultDelegateUser());
 
             //When
-            var returnedDelegateUsers = userService.GetDelegateUsersByUsername("SV1234");
-            returnedDelegateUsers.First().Id = expectedDelegateUsers.Id;
+            var (returnedAdminUser, _) = userService.GetUsersById("1", "2");
+            var (_, returnedDelegateUser) = userService.GetUsersById("1", "2");
```

Comment:
Could we just do var (returnedAdminUser, returnedDelegateUser) and call userService once?

Reply:
Yeah probably, I just copied the pattern from the other existing tests I refactored slightly